### ISSUE TITLE
Eliminate unavailable combinations from product page

### ISSUE THREAD

**Issue by marktnz1** [thirtybees/niara#61]
Users trying to choose combinations on the product page currently get presented with all created combinations for that product. They should only be shown those combinations that are available for that product and not receive a "This combination does not exist for this product. Please select another combination" warning.

**Comment by getdatakick** [comment]
This is a tricky on, and requires a very good UX design.

The problem is that there can be multiple attribute input selectors. To ensure that only valid combination can be selected, page javascript would have to filter available attribute options every time some other attribute selector changes. But that can have negative impact on user experience.

Consider this interaction:
1) User starts by selecting blue color
2) after color selection, size attribute options was filtered, and option 'XL' was removed from list (because combination Blue,XL does not exists). 
3) unfortunately, user needs XL, and they don't really care that much about the color. Howerver, because of the color selection, they are not able to choose XL size, even if it exists in other combination
4) they first need to got back to color, and change it something else, and hope that XL size will be available for that

That's not very user friendly. 

We would probably have to allow some on-page faceted combination search.

**Comment by marktnz1** [comment]
Yes true, that would be both perfect and difficult...  plus all the warehousing options if ASM is selected or not and optionally display available quantities of each combo.

There are available modules on "the market" that claim to deal with this but I do think this should be core function.

**Comment by getdatakick** [comment]
Bug is when something does not work as intended. This was designed this way, and works properly. It can be improved to make a better user experience, hence the enhancement.

**Comment by marktnz1** [comment]
![Screenshot from 2022-12-10 16-09-56](https://user-images.githubusercontent.com/25965650/206826343-c786f9fe-fe02-425a-a6aa-68e809f141ef.png)

The feature specifically says "Display unavailable product attributes on the product page" Ive set it to NO. This means literally "DO NOT display unavailable product attributes on the product page". If this was designed to do something else, I suggest you clarify on the site what the function is, because to me there's no argument at all, the function is NOT working as described since unavailable attributes are being displayed despite me switching that OFF.

**Comment by getdatakick** [comment]
This option works on attributes, not on combinations, and is based on available quantity.

With this option disabled:
For any attribute value, if all combination using this attribute value are not available (0 quantity on stock), then this attribute will be hidden. This is kind of pre-filtering.

With this option enable:
all attribute values (used by product) will be always displayed

This works as designed. 

This is just a static optimization to limit the amount of combinations that user can choose from. But they still can choose invalid combination.
